File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        ways = [1, 2]

        for i in range(2, n):
            temp = ways[0] + ways[1]
            ways[0], ways[1] = ways[1], temp

        return ways[-1] if n > 1 else ways[0]

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int a = 1;
        int b = 2;
        for (int i = 2; i < n; ++i) {
            const int temp = a + b;
            a = b;
            b = temp;
        }
        return n == 1 ? a : b;
    }
};